gateway-core/components/org.wso2.carbon.gateway.core/src/main/java/org/wso2/carbon/gateway/core/config/dsl/external/wuml/WUMLBaseListenerImpl.java

```diff
@@ -675,54 +666,139 @@ public void exitParExpression(WUMLParser.ParExpressionContext ctx) {
 
                 Condition condition = new Condition(source, Pattern.compile(conditionValue));
 
-                filterMediator = new FilterMediator(condition);
+                FilterMediator filterMediator = new FilterMediator(condition);
+                dropMediatorFilterAware(filterMediator);
+                flowControllerStack.push(filterMediator);
+                this.flowControllerMediatorSection.push(FlowControllerMediatorSection.ifBlock);
             }
         }
     }
 
     @Override
     public void exitIfElseBlock(WUMLParser.IfElseBlockContext ctx) {
-        this.elseBlockOpened = false;
-        this.ifBlockOpened = false;
-        dropMediatorFilterAware(filterMediator);
+        this.flowControllerStack.pop();
     }
 
     @Override
     public void enterIfBlock(WUMLParser.IfBlockContext ctx) {
-        this.elseBlockOpened = false;
-        this.ifBlockOpened = true;
     }
 
     @Override
     public void exitIfBlock(WUMLParser.IfBlockContext ctx) {
-        this.ifBlockOpened = false;
+        this.flowControllerMediatorSection.pop();
     }
 
     @Override
     public void enterElseBlock(WUMLParser.ElseBlockContext ctx) {
-        this.ifBlockOpened = false;
-        this.elseBlockOpened = true;
+        this.flowControllerMediatorSection.push(FlowControllerMediatorSection.elseBlock);
     }
 
     @Override
     public void exitElseBlock(WUMLParser.ElseBlockContext ctx) {
+        this.flowControllerMediatorSection.pop();
     }
 
+
+    /* Try-catch parsing */
+    /**
+     * {@inheritDoc}
+     *
+     * <p>The default implementation does nothing.</p>
+     */
+    @Override public void enterTryClause(WUMLParser.TryClauseContext ctx) {
+        TryBlockMediator tryBlockMediator = new TryBlockMediator();
+        dropMediatorFilterAware(tryBlockMediator);
+        flowControllerStack.push(tryBlockMediator);
+        this.flowControllerMediatorSection.push(FlowControllerMediatorSection.tryBlock);
+    }
+    /**
+     * {@inheritDoc}
+     *
+     * <p>The default implementation does nothing.</p>
+     */
+    @Override public void exitTryClause(WUMLParser.TryClauseContext ctx) {
+        this.flowControllerMediatorSection.pop();
+    }
+    /**
+     * {@inheritDoc}
+     *
+     * <p>The default implementation does nothing.</p>
+     */
+    @Override public void exitExceptionHandler(WUMLParser.ExceptionHandlerContext ctx) {
+
+        if (ctx.getChild(0) != null && ctx.getChild(0).getChild(0) != null) {
+
+            String exceptionType = ((java.util.ArrayList) ((WUMLParser.ExceptionTypeContext) ctx.children
+                    .get(0)).children).get(0).toString();
```

Comment:
Thanks for the idea, I have included a comment with https://github.com/wso2/carbon-gateway-framework/pull/40